src/NodeChanger/MethodNameChanger.php

```diff
@@ -2,21 +2,23 @@
 
 namespace Rector\NodeChanger;
 
-use PhpParser\Node;
-use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr;
 use PhpParser\Node\Identifier;
 
 final class MethodNameChanger
 {
+    public function renameNode(Expr $node, string $newMethodName): void
```

Comment:
@TomasVotruba I could force it to be an `Expr`. Some places, `Rector/Dynamic/PropertyNameReplacerRector` for example, passes a `PropertyFetch` to `renameNode`. What should we do?

Reply:
Well, this is `MethodNameChanger`, so it takes care of `MethodCall`.

`PropertyFetch` should be handled by `PropertyNameChanger` class.

It's similar to [NodeAnalyzers](https://github.com/rectorphp/rector/tree/master/src/NodeAnalyzer), where each node has own service that has own rules for them.

So I'd create `PropertyNameChanger`, if I guessed that corretly.

Reply:
Hm, one more `Changer`, got! Gonna create it.

If an abstraction arise, I'll handle it :man_technologist: